pkg/northbound/gnmi/set.go

```diff
@@ -187,15 +188,19 @@ func (s *Server) Set(ctx context.Context, req *gnmi.SetRequest) (*gnmi.SetRespon
 		return nil, status.Error(codes.Internal, errSet.Error())
 	}
 	if userName != "" {
+		s.mu.Lock()
 		ch, err := mgr.NetworkChangesStore.Get(change.ID)
 		if err != nil {
-			return nil, status.Error(codes.Internal, errSet.Error())
+			s.mu.Unlock()
+			return nil, errors.NewInternal("Error getting NetworkChange %s: %s", change.ID, err.Error())
```

Comment:
FYI the reason this `Internal` error is causing an `Unknown` error is because this is a gRPC server and the error is not being converted for gRPC. If you do `errors.Status(err).Err()` you’ll get a gRPC error that will return the correct error code to the client.

Reply:
Also, keep in mind Atomix returns a lot more rich error information now — 'NotFound` when a key is not in the store, `AlreadyExists' when a key is already in the store, `Conflict` when there’s an update conflict, `Internal` when there’s an error inside the state machine, etc — and when we wrap errors in `NewInternal` we’re losing that type information. We really are trying to be more precise in which errors we want to propagate up to the client and which we can handle internally, as you can see in the E2T snippet above. That said, if this _is_ an error we want to propagate up to the client, I think calling it an `Internal` error is fine.

Reply:
good to know - thanks